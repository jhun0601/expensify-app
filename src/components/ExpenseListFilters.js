import React from "react";

import "react-dates/initialize";
// import "react-dates/lib/css/_datepicker.css";
import { DateRangePicker } from "react-dates";

import { connect } from "react-redux";
import {
    setTextFilter,
    sortByDate,
    sortByAmount,
    setStartDate,
    setEndDate,
} from "../actions/filters";

export class ExpenseListFilters extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            calendarFocused: null,
        };

        this.onDatesChange = this.onDatesChange.bind(this);
        this.onFocusChange = this.onFocusChange.bind(this);
    }

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    };
    onSortChange = (e) => {
        if (e.target.value === "date") {
            this.props.sortByDate();
        } else if (e.target.value === "amount") {
            this.props.sortByAmount();
        }
    };
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            type="text"
                            className="text-input"
                            placeholder="Search expenses"
                            value={this.props.filters.text}
                            onChange={this.onTextChange}
                        />
                    </div>
                    <div className="input-group__item">
                        <select
                            value={this.props.filters.sortBy}
                            className="select"
                            onChange={this.onSortChange}
                        >
                            <option value="amount">Amount</option>
                            <option value="date">Date</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange}
                            showClearDates={true}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                            startDateId={"3dfjklds"}
                            endDateId={"asdasf"}
                        />
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters,
});
const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDate: () => dispatch(sortByDate()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
});
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
