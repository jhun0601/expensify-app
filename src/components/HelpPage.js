import React from "react";

const HelpPage = () => <div>This is a Help Page.</div>;

export default HelpPage;
