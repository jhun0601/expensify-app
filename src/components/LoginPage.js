import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";
export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__container">
            <img
                src="/images/favicon.png"
                alt="expensify app"
                className="box-layout__logo"
            />
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button onClick={startLogin} className="button">
                Log in with Google
            </button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin()),
});
export default connect(undefined, mapDispatchToProps)(LoginPage);
