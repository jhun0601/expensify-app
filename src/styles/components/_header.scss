.header {
    background: $dark-blue;
    &__content {
        display: flex;
        justify-content: space-between;
        align-items: center;
        padding: $s-size 0;
    }
    &__title {
        color: #fff;
        text-decoration: none;
        h1 {
            margin: 0;
        }
    }
}
