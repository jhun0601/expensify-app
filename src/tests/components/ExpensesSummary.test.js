import React from "react";
import { shallow } from "enzyme";

import { ExpensesSummary } from "../../components/ExpensesSummary";

test("Should correctly render ExpenseSummary with 1 expense", () => {
    const wrapper = shallow(
        <ExpensesSummary expenseCount={1} expensesTotal={1316546} />
    );
    expect(wrapper).toMatchSnapshot();
});

test("Should correctly render ExpenseSummary with multiple expenses", () => {
    const wrapper = shallow(
        <ExpensesSummary expenseCount={1} expensesTotal={1316546} />
    );
    expect(wrapper).toMatchSnapshot();
});
