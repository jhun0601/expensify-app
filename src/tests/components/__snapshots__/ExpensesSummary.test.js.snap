// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $13,165.46
  </h1>
</div>
`;

exports[`Should correctly render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $13,165.46
  </h1>
</div>
`;
